users/permissions.py

```diff
@@ -30,26 +30,48 @@ def has_permission(self, request, view):
         if request.user.is_superuser:
             return True
 
-        if view.action == "create":
-            # only organization admins can add a user to their organization
-            user_is_organization_admin = OrganizationUser.objects.filter(
-                organization=request.data["organization"],
-                user=request.user.id,
-                role__name="org-admin",
-            ).exists()
-            return user_is_organization_admin
+        # when listing, a user should only see organization-user mapping for the organizations they have access to.
+        # this is handled by get_queryset
+        if view.action == "list":
+            return True
 
-        return True
+        user_organization_role = request.user.get_role_for_organization(
+            request.data["organization"]
+        )
+        if user_organization_role.name not in ["org-admin", "super-admin"]:
+            # user doesn't belong to the queried organization
+            # or doesn't have sufficient role within organization
+            return False
+
+        requested_role = Role.objects.filter(id=request.data["role"]).first()
```

Comment:
what is request.data doesn't have any role value?

Reply:
what if the provided role Id is wrong?

Reply:
I thought about it but felt like a role must be a required value in this query. I'm not sure in which case it won't be provided.

I agree on incorrect role id. That check needs to be in place. I'll see what's the right fix.